WordPress/gutenberg
# `NavigableContainer`: convert to TypeScript

## What?
Convert `NavigableContainer` components to TypeSccript

## Why?
As part of https://github.com/WordPress/gutenberg/issues/35744

## How?
Mainly by adding types

## Testing Instructions
1. `npm run storybook:dev`
2. Go to the stories for `NavigableMenu` and `TabbableContainer`
3. Expected: They work like before:
<img width="1064" alt="Screenshot 2023-04-01 at 5 15 53 PM" src="https://user-images.githubusercontent.com/4063887/229320794-340f5b45-c188-4ca8-aa5a-090c7c11cf70.png">

### Testing Instructions for Keyboard
1. `npm run storybook:dev`
2. Go to the stories for `NavigableMenu` and `TabbableContainer`
3. Use the keyboard to tab through the components
5. Expected: The work like before

## Screenshots or screencast 
See above
